rir/src/ir/Compiler.cpp

```diff
@@ -820,8 +820,9 @@ void compileCall(Context& ctx, SEXP ast, SEXP fun, SEXP args) {
 
         // (1) Arguments are wrapped as Promises:
         //     create a new Code object for the promise
-        size_t prom = compilePromise(ctx, *arg);
-        callArgs.push_back(prom);
+        Code* prom = compilePromise(ctx, *arg);
```

Comment:
@o- I think here and then later down with the default args, these promises can get collected before they are written into the extra pool.